example/test-pipeline.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: numaflow.numaproj.io/v1alpha1
+kind: Pipeline
+metadata:
+  name: nats-source-e2e
+spec:
+  vertices:
+    - name: in
+      scale:
+        min: 2
+      volumes:
+        - name: my-config-mount
+          configMap:
+            name: nats-config-map
+        - name: my-secret-mount
+          secret:
+            secretName: nats-auth-fake-token
+      source:
+        udsource:
+          container:
+            image: quay.io/numaio/numaflow-source/nats-source:v0.5.2
+            volumeMounts:
+              - name: my-config-mount
+                mountPath: /etc/config
+              - name: my-secret-mount
+                mountPath: /etc/secrets/nats-auth-fake-token
+    - name: out
+      sink:
+        log: {}
+  edges:
+    - from: in
+      to: out
```

Comment:
One thing to note is I guess if the Pipeline Spec ever changes in a way that's not backward compatible with this we will need to remember to change this and the README

Reply:
Now that you mention it, having tests that truly test out the interaction with the real source is actually critical to our being able to rely on the dependabot to automatically update dependencies. We need to consider if it's better that an e2e test actually use numaflow or just be end-to-end within the application itself. My original thought was the latter. This was what I wrote [here](https://docs.google.com/document/d/1N_UdwLWGMJjIZKbMsTZfUTTzNI1Pff9oLYuuB-eyvdo/edit):
`
What should our tests consist of? The objective of the tests will be to:
sufficiently test the user’s code
sufficiently test the use of the dependency in the code
Should we use an e2e or Unit test? Theoretically, if we want to keep it simple, a unit test might actually suffice as long as we exercise each of the main interface methods, and we don’t stub out the dependencies. But if we think that changes the definition of “Unit test”, then we can look into the best method of using an e2e test:
Option 1: e2e test using Numaflow pipelines
Option 2: “e2e” really just means the end to end for this Container itself

Option 1 seems simpler to do since we already have e2e tests executing pipelines that we could imitate, but it seems like it might be the wrong approach, as it depends on numaflow itself. Say that we always test with main:latest, and main:latest is broken in some way on numaflow. Then these tests will fail. 
`

Reply:
We can start with writing good unit tests with high coverage and evaluate the necessity of adding E2E test with real numaflow pipeline as we collect more data points.

Reply:
right, I think that can work